src/common/mi-lttng.c

```diff
@@ -1208,6 +1210,141 @@ int mi_lttng_event_function_probe(struct mi_writer *writer,
 	return ret;
 }
 
+LTTNG_HIDDEN
+int mi_lttng_event_userspace_probe(struct mi_writer *writer,
+		struct lttng_event *event)
+{
+	int ret;
+	struct lttng_userspace_probe_location *location;
+	struct lttng_userspace_probe_location_lookup_method *lookup_method;
+	enum lttng_userspace_probe_location_lookup_method_type lookup_type;
+
+	location = lttng_event_get_userspace_probe_location(event);
+	if (!location) {
+		goto end;
+	}
+
+	lookup_method = lttng_userspace_probe_location_get_lookup_method(location);
+	if (!lookup_method) {
+		goto end;
+	}
+
+	lookup_type = lttng_userspace_probe_location_lookup_method_get_type(lookup_method);
+
+	ret = mi_lttng_writer_open_element(writer, config_element_attributes);
+	if (ret) {
+		goto end;
+	}
+
+	switch (lttng_userspace_probe_location_get_type(location)) {
+	case LTTNG_USERSPACE_PROBE_LOCATION_TYPE_FUNCTION:
+	{
+		const char *function_name;
+		const char *binary_path;
+		ret = mi_lttng_writer_write_element_string(writer,
+					config_element_userspace_probe_type,
+					config_element_userspace_probe_type_function);
+		if (ret) {
+			goto end;
+		}
+		binary_path = lttng_userspace_probe_location_function_get_binary_path(location);
+		ret = mi_lttng_writer_write_element_string(writer,
+					config_element_userspace_probe_binary_path, binary_path);
+		if (ret) {
+			goto end;
+		}
+		switch (lookup_type) {
+		case LTTNG_USERSPACE_PROBE_LOCATION_LOOKUP_METHOD_TYPE_FUNCTION_ELF:
+			ret = mi_lttng_writer_write_element_string(writer,
+						config_element_userspace_probe_lookup,
+						config_element_userspace_probe_lookup_elf);
```

Comment:
Remove one indentation level. The same applies for other parts of the function.

Reply:
done